plugin/storage/es/spanstore/dbmodel/to_domain.go

```diff
@@ -176,6 +174,18 @@ func (td ToDomain) convertTagField(k string, v interface{}) (model.KeyValue, err
 	// the binary is never returned, ES returns it as string with base64 encoding
 	case []byte:
 		return model.Binary(dKey, val), nil
+	// in spans are decoded using json.UseNumber() to preserve the type
+	// however note that float(1) will be parsed as int as ES does not store decimal point
+	case json.Number:
+		n, err := val.Int64()
+		if err == nil {
+			return model.Int64(dKey, n), nil
+		}
+		f, err := val.Float64()
+		if err == nil {
+			return model.Float64(dKey, f), nil
+		}
+		return model.String("", ""), err
```

Comment:
Would it be better to return an error similar to the next line, to provide more context?

Reply:
+1 I will wrap it